### DISCUSSION TITLE
Display format

### DISCUSSION THREAD

**Discussion by huddledfiber** [sqlitebrowser/sqlitebrowser#3014] [Q&A]
I'm trying to edit the display format for a date column that contains nanoseconds since UTC 2001.  A colleague had given me this but it isn't working.
strftime('%m-%d-%Y %H:%M:%f', CAST("date" AS FLOAT) / 1000

If possible I'd like to display it with UTC-6 taken into account.

**Comment by chrisjlocke** [comment]
> but it isn't working.

Have you got an example of the data you're using, and the output currently being produced?
Is it displaying anything?  Or just the wrong value?

**Reply by huddledfiber** [reply]
Here's one of the values I'm trying to convert.  I know the correct date is 4/25/2021 8:41:47 PM
![2022-04-13 19_02_55-DB Browser for SQLite - Z__FB-22-0033_For Examiner_sms db](https://user-images.githubusercontent.com/62117460/163289038-6cf29567-626b-44f6-a591-56be335e9ed9.jpg)
.

**Reply by huddledfiber** [reply]
I think the value got wiped.  This is the value. 641076107244000000

**Reply by chrisjlocke** [reply]
Aah, looks like you're missing a closing bracket.  Brackets have to match - a closing bracket for each opening bracket.

DB4S actually highlights this if you place the cursor after the first closing bracket - highlighting which bracket its closing.

![image](https://user-images.githubusercontent.com/7872218/163327060-baf3fc73-04ca-4cd8-9aa5-6a17d5b6eea0.png)

So I added a closing one, and it didn't work 

![image](https://user-images.githubusercontent.com/7872218/163327807-65ae1029-8f4d-4086-9d89-551e7d958f5d.png)

But bear in mind strftime expects a number based on the UNIX epoch, so it's expecting ~1649920282, which it can convert.  DB4S has an option for this

![image](https://user-images.githubusercontent.com/7872218/163333156-3f91a304-d85e-4bc6-8166-d48ed9923e5c.png)

![image](https://user-images.githubusercontent.com/7872218/163333236-80ecbccc-ecd2-4b67-b727-7eaaea05c2af.png)

![image](https://user-images.githubusercontent.com/7872218/163333268-aedd70d1-32d1-4765-bfe0-7d521e46e17d.png)

or the Java version

![image](https://user-images.githubusercontent.com/7872218/163334205-fd6b05e8-286e-45a8-921c-f9581cc225b4.png)

which is similar to what you're doing.
However, using strftime will get you the wrong date (dividing by 1000000 to get to around ~1649920282)

![image](https://user-images.githubusercontent.com/7872218/163335046-f53ac3ef-6d0a-4dca-b5f8-657e305e238e.png)

As it's based on the Unix epoch, not the Apple one, which is based on 1 Jan 2001.  So if you add 978307200 seconds, you'll convert the Cocoa Data Time format (that's another name for it) to a 'proper' format...

![image](https://user-images.githubusercontent.com/7872218/163335665-e6f1c6a4-0b29-453c-8913-d7c41fc4d684.png)

So note the date is now 25 Apr.  Voila.

**Reply by huddledfiber** [reply]
That worked! Is there a way to make it display in UTC-6 or UTC-5?

**Comment by huddledfiber** [comment]
That worked!  Is there a way to make it display in UTC-6 or UTC-5?

**Reply by chrisjlocke** [reply]
Forgive my ignorance, but I don't understand the question.  UTC-6 is a timezone?  So if the data is 10:00, you want to display (for example) five hours ahead of that, so 15:00 ?  If its a set number of hours, just add that in the query ... so if you want all the time + 5 hours (18000 seconds), you'd put the display format as "(field2"/1000000000)+978307200+18000"

**Comment by huddledfiber** [comment]
I got it now, thank you!
